selfdrive/ui/sunnypilot/qt/offroad/settings/vehicle_panel.cc

```diff
@@ -20,11 +20,101 @@ VehiclePanel::VehiclePanel(QWidget *parent) : QFrame(parent) {
   platformSelector = new PlatformSelector();
   list->addItem(platformSelector);
 
+//   hkgtuningToggle = new ParamControl("HKGtuning", (add all toggle parameters here), this);
+
+  QObject::connect(uiState(), &UIState::offroadTransition, [=](bool offroad) {
+    isOnroad = !offroad;
+    updateCarToggles();
+  });
+
   ScrollViewSP *scroller = new ScrollViewSP(list, this);
   vlayout->addWidget(scroller);
 
   QObject::connect(uiState(), &UIState::offroadTransition, platformSelector, &PlatformSelector::refresh);
 
+  QObject::connect(platformSelector, &PlatformSelector::clicked, [=]() {
+    QTimer::singleShot(100, this, &VehiclePanel::updateCarToggles);
+  });
+
+  QTimer *timer = new QTimer(this);
+  QObject::connect(timer, &QTimer::timeout, this, &VehiclePanel::updateCarToggles);
+  timer->start(1000);
+
   main_layout->addWidget(vehicleScreen);
   main_layout->setCurrentWidget(vehicleScreen);
 }
+
+void VehiclePanel::showRebootPrompt() {
+  QString question = tr("Changing this setting requires a reboot to take effect.\n\n"
+                       "Reboot now or later?");
+
+  reboot_confirmed = ConfirmationDialog::confirm(question, tr("Reboot Now"), this);
+  if (reboot_confirmed && !reboot_completed) {
+    params.putBool("DoReboot", true);
+    reboot_completed = true;
+  }
+}
```

Comment:
For most options and toggle changes, we usually shouldn't need to perform a full reboot to take effect. How about if we invoke "Always Offroad" mode we have in sunnypilot? We introduced "Offroad Mode" in sunnypilot a little while back. This practically allows the settings to take effect without a full reboot:
- https://github.com/sunnypilot/sunnypilot/pull/596